{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a140b78d-af99-4bb1-bdc4-13a83b06cdc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fd74550-68ee-4f5b-bfe7-bad669c6a2eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cc38e25f-4de7-437f-a7d8-7638b72a7b57",
   "metadata": {},
   "outputs": [],
   "source": [
    "#################################### READ IN AND COMBINE TABLES ####################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bfe458c-90f7-4bff-a2ad-0c2a38dc9331",
   "metadata": {},
   "outputs": [],
   "source": [
    "###\n",
    "### For release2016 table ###\n",
    "annotations_table = pd.read_csv('/omics/groups/OE0606/internal/shared/icgc_pcawg/annotations/summary_table_combined_annotations_v3.txt', sep='\\t')\n",
    "annotations_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8df5a8b6-3f9b-4add-9f7a-4618b84dba62",
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations_table.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "1224f6e3-bb7c-4e83-a5df-70d282a79704",
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations_table_small = annotations_table[['icgc_donor_id', 'icgc_sample_id','samplename','purity', 'histology_abbreviation', 'cancer_type']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "60cf7330-c5a8-42eb-8788-5739f3b1a38e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>samplename</th>\n",
       "      <th>purity</th>\n",
       "      <th>histology_abbreviation</th>\n",
       "      <th>cancer_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DO46416</td>\n",
       "      <td>SA505245</td>\n",
       "      <td>0009b464-b376-4fbc-8a56-da538269a02f</td>\n",
       "      <td>0.885</td>\n",
       "      <td>Ovary-AdenoCA</td>\n",
       "      <td>OV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DO36062</td>\n",
       "      <td>SA413814</td>\n",
       "      <td>003819bc-c415-4e76-887c-931d60ed39e7</td>\n",
       "      <td>0.774</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>PBCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DO45049</td>\n",
       "      <td>SA501385</td>\n",
       "      <td>0040b1b6-b07a-4b6e-90ef-133523eaf412</td>\n",
       "      <td>0.800</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>LINC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_donor_id icgc_sample_id                            samplename  purity  \\\n",
       "0       DO46416       SA505245  0009b464-b376-4fbc-8a56-da538269a02f   0.885   \n",
       "1       DO36062       SA413814  003819bc-c415-4e76-887c-931d60ed39e7   0.774   \n",
       "2       DO45049       SA501385  0040b1b6-b07a-4b6e-90ef-133523eaf412   0.800   \n",
       "\n",
       "  histology_abbreviation cancer_type  \n",
       "0          Ovary-AdenoCA          OV  \n",
       "1          CNS-PiloAstro        PBCA  \n",
       "2              Liver-HCC        LINC  "
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations_table_small.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "113c18dc-93a7-4666-8aad-684ce31a5acf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2834, 91)\n"
     ]
    }
   ],
   "source": [
    "###\n",
    "### Now for release2016 table ###\n",
    "\n",
    "release_may2016_table = pd.read_csv('/omics/groups/OE0606/internal/shared/icgc_pcawg/annotations/release_may2016.v1.4.tsv', sep='\\t')\n",
    "print(release_may2016_table.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "211e382a-ca96-4956-bdc9-08ae12f72c5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#release_may2016_table.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "9fc03ffb-4d33-4732-a684-a1806e7e4855",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>donor_unique_id</th>\n",
       "      <th>wgs_exclusion_white_gray</th>\n",
       "      <th>dcc_project_code</th>\n",
       "      <th>submitter_donor_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>mar2016_donor</th>\n",
       "      <th>santa_cruz_pilot</th>\n",
       "      <th>validation_by_deep_seq</th>\n",
       "      <th>TiN</th>\n",
       "      <th>normal_wgs_submitter_specimen_id</th>\n",
       "      <th>normal_wgs_icgc_specimen_id</th>\n",
       "      <th>normal_wgs_submitter_sample_id</th>\n",
       "      <th>normal_wgs_icgc_sample_id</th>\n",
       "      <th>normal_wgs_aliquot_id</th>\n",
       "      <th>normal_wgs_bwa_alignment_gnos_repo</th>\n",
       "      <th>normal_wgs_bwa_alignment_gnos_id</th>\n",
       "      <th>is_mar2016_normal_wgs_bwa_alignment</th>\n",
       "      <th>normal_wgs_bwa_alignment_bam_file_name</th>\n",
       "      <th>normal_wgs_minibam_gnos_repo</th>\n",
       "      <th>normal_wgs_minibam_gnos_id</th>\n",
       "      <th>is_mar2016_normal_wgs_minibam</th>\n",
       "      <th>normal_wgs_minibam_bam_file_name</th>\n",
       "      <th>tumor_wgs_specimen_count</th>\n",
       "      <th>tumor_wgs_submitter_specimen_id</th>\n",
       "      <th>tumor_wgs_icgc_specimen_id</th>\n",
       "      <th>tumor_wgs_submitter_sample_id</th>\n",
       "      <th>tumor_wgs_icgc_sample_id</th>\n",
       "      <th>tumor_wgs_aliquot_id</th>\n",
       "      <th>tumor_wgs_oxog_score</th>\n",
       "      <th>tumor_wgs_ContEST</th>\n",
       "      <th>tumor_wgs_Stars</th>\n",
       "      <th>tumor_wgs_bwa_alignment_gnos_repo</th>\n",
       "      <th>tumor_wgs_bwa_alignment_gnos_id</th>\n",
       "      <th>is_mar2016_tumor_wgs_bwa_alignment</th>\n",
       "      <th>tumor_wgs_bwa_alignment_bam_file_name</th>\n",
       "      <th>tumor_wgs_minibam_gnos_repo</th>\n",
       "      <th>tumor_wgs_minibam_gnos_id</th>\n",
       "      <th>is_mar2016_tumor_wgs_minibam</th>\n",
       "      <th>tumor_wgs_minibam_bam_file_name</th>\n",
       "      <th>sanger_variant_calling_repo</th>\n",
       "      <th>sanger_variant_calling_gnos_id</th>\n",
       "      <th>sanger_variant_calling_file_name_prefix</th>\n",
       "      <th>is_mar2016_sanger_variant_calling</th>\n",
       "      <th>sanger_variant_calling_deprecated_gnos_id</th>\n",
       "      <th>dkfz_embl_variant_calling_repo</th>\n",
       "      <th>dkfz_embl_variant_calling_gnos_id</th>\n",
       "      <th>dkfz_embl_variant_calling_file_name_prefix</th>\n",
       "      <th>is_mar2016_dkfz_embl_variant_calling</th>\n",
       "      <th>dkfz_embl_variant_calling_deprecated_gnos_id</th>\n",
       "      <th>broad_variant_calling_repo</th>\n",
       "      <th>broad_variant_calling_gnos_id</th>\n",
       "      <th>broad_variant_calling_file_name_prefix</th>\n",
       "      <th>is_mar2016_broad_variant_calling</th>\n",
       "      <th>broad_variant_calling_deprecated_gnos_id</th>\n",
       "      <th>muse_variant_calling_repo</th>\n",
       "      <th>muse_variant_calling_gnos_id</th>\n",
       "      <th>muse_variant_calling_file_name_prefix</th>\n",
       "      <th>is_mar2016_muse_variant_calling</th>\n",
       "      <th>broad_tar_variant_calling_repo</th>\n",
       "      <th>broad_tar_variant_calling_gnos_id</th>\n",
       "      <th>broad_tar_variant_calling_file_name_prefix</th>\n",
       "      <th>is_mar2016_broad_tar_variant_calling</th>\n",
       "      <th>normal_wgs_has_matched_rna_seq</th>\n",
       "      <th>tumor_wgs_has_matched_rna_seq</th>\n",
       "      <th>normal_rna_seq_submitter_specimen_id</th>\n",
       "      <th>normal_rna_seq_icgc_specimen_id</th>\n",
       "      <th>normal_rna_seq_submitter_sample_id</th>\n",
       "      <th>normal_rna_seq_icgc_sample_id</th>\n",
       "      <th>normal_rna_seq_aliquot_id</th>\n",
       "      <th>normal_rna_seq_star_alignment_gnos_repo</th>\n",
       "      <th>normal_rna_seq_star_alignment_gnos_id</th>\n",
       "      <th>is_mar2016_normal_rna_seq_star_alignment</th>\n",
       "      <th>normal_rna_seq_star_alignment_bam_file_name</th>\n",
       "      <th>normal_rna_seq_tophat_alignment_gnos_repo</th>\n",
       "      <th>normal_rna_seq_tophat_alignment_gnos_id</th>\n",
       "      <th>is_mar2016_normal_rna_seq_tophat_alignment</th>\n",
       "      <th>normal_rna_seq_tophat_alignment_bam_file_name</th>\n",
       "      <th>tumor_rna_seq_specimen_count</th>\n",
       "      <th>tumor_rna_seq_submitter_specimen_id</th>\n",
       "      <th>tumor_rna_seq_icgc_specimen_id</th>\n",
       "      <th>tumor_rna_seq_submitter_sample_id</th>\n",
       "      <th>tumor_rna_seq_icgc_sample_id</th>\n",
       "      <th>tumor_rna_seq_aliquot_id</th>\n",
       "      <th>tumor_rna_seq_star_alignment_gnos_repo</th>\n",
       "      <th>tumor_rna_seq_star_alignment_gnos_id</th>\n",
       "      <th>is_mar2016_tumor_rna_seq_star_alignment</th>\n",
       "      <th>tumor_rna_seq_star_alignment_bam_file_name</th>\n",
       "      <th>tumor_rna_seq_tophat_alignment_gnos_repo</th>\n",
       "      <th>tumor_rna_seq_tophat_alignment_gnos_id</th>\n",
       "      <th>is_mar2016_tumor_rna_seq_tophat_alignment</th>\n",
       "      <th>tumor_rna_seq_tophat_alignment_bam_file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BLCA-US::096b4f32-10c1-4737-a0dd-cae04c54ee33</td>\n",
       "      <td>Whitelist</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>096b4f32-10c1-4737-a0dd-cae04c54ee33</td>\n",
       "      <td>DO804</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>0.01</td>\n",
       "      <td>27461a27-26eb-4c2c-9c54-e16fbd32c615</td>\n",
       "      <td>SP1682</td>\n",
       "      <td>e0fccaf5-925a-41f9-b87c-cd5ee4aecb59</td>\n",
       "      <td>SA5237</td>\n",
       "      <td>e0fccaf5-925a-41f9-b87c-cd5ee4aecb59</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>f22a72c5-73c8-478d-b03e-04599b9d5321</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.67455c36-aa47-4cc4-8b6d-9a9012b616ed.bam</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>7e442040-f6ea-11e5-bd75-997f5888a0e8</td>\n",
       "      <td>False</td>\n",
       "      <td>PCAWG.67455c36-aa47-4cc4-8b6d-9a9012b616ed_min...</td>\n",
       "      <td>1</td>\n",
       "      <td>52f538ef-b05d-4c76-9976-ce6d49158016</td>\n",
       "      <td>SP1677</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>SA5195</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>87.52</td>\n",
       "      <td>0.1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>a6e8dd23-c8a5-445a-ae4b-b9f92ed6a73e</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.a833602e-7ee7-442b-9bf9-f6489cd64ba2.bam</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>7e442040-f6ea-11e5-bd75-997f5888a0e8</td>\n",
       "      <td>False</td>\n",
       "      <td>PCAWG.a833602e-7ee7-442b-9bf9-f6489cd64ba2_min...</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>8935fe36-ceec-11e5-bedc-767d4d100685</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>True</td>\n",
       "      <td>5230346f-987e-44c3-b9fe-dddb5829e247</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>4269588d-5012-4eee-ab9a-9c48897dae69</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>ac34e35d-21b1-4b11-b42d-155cba1a547f</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>True</td>\n",
       "      <td>d12904ff-4ce6-4d7b-a0d7-69e18a3822cc</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>48077833-99ed-49c6-88a1-80cd2e50755a</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>True</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>38ce6165-c324-4642-89cf-984d738ee08a</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52f538ef-b05d-4c76-9976-ce6d49158016</td>\n",
       "      <td>SP1677</td>\n",
       "      <td>22e154de-0e3b-443b-8420-48d68d6c1ce4</td>\n",
       "      <td>SA5213</td>\n",
       "      <td>22e154de-0e3b-443b-8420-48d68d6c1ce4</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>a63c9c0a-c8aa-44dc-8915-08f55109a5c9</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.f4288632-a94f-4777-a5d6-1a76b12abcc5.STA...</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>81438f7c-ee87-44a7-acd1-562ec8f386f0</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.f4288632-a94f-4777-a5d6-1a76b12abcc5.Top...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BLCA-US::178b28cd-99c3-48dc-8d09-1ef71b4cee80</td>\n",
       "      <td>Whitelist</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>178b28cd-99c3-48dc-8d09-1ef71b4cee80</td>\n",
       "      <td>DO555</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>0.00</td>\n",
       "      <td>cd3cfb26-e66f-408e-81f6-3b61c247c976</td>\n",
       "      <td>SP1135</td>\n",
       "      <td>c1da8eed-4919-4ba5-a735-3fba476c18a7</td>\n",
       "      <td>SA1598</td>\n",
       "      <td>c1da8eed-4919-4ba5-a735-3fba476c18a7</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>964bac8f-43cc-4a47-b233-53798d287029</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.13985b6b-d7b6-4efd-9c53-288f4d269a0c.bam</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>0b6a5d16-e617-11e5-ad33-d98536d081ee</td>\n",
       "      <td>False</td>\n",
       "      <td>PCAWG.13985b6b-d7b6-4efd-9c53-288f4d269a0c_min...</td>\n",
       "      <td>1</td>\n",
       "      <td>59d6683f-5eb7-493d-8e8e-78b88be2cd70</td>\n",
       "      <td>SP1132</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>SA1556</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>54.55</td>\n",
       "      <td>0.1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>437e11a0-4137-4614-9f64-c5e798c8bb33</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.af830047-822a-4dcc-88bc-68f7440a208d.bam</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>0b6a5d16-e617-11e5-ad33-d98536d081ee</td>\n",
       "      <td>False</td>\n",
       "      <td>PCAWG.af830047-822a-4dcc-88bc-68f7440a208d_min...</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>714e7b6a-d34a-11e5-90f0-f8135c6878de</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>True</td>\n",
       "      <td>c0387aa4-9937-45f7-bf3a-e2da65457738</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>875fdb54-76ae-11e5-abee-bff16f24f4a1</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>d989e42d-354c-4652-9921-27f79ba31c2e</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>True</td>\n",
       "      <td>f3c6d06b-09ff-4e0d-821d-80bd2b04106f</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>d9ec79b7-0ccb-4312-b84d-b90200eb77bf</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>True</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>9f6910ec-80dc-4270-b415-26c00f5ad281</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>59d6683f-5eb7-493d-8e8e-78b88be2cd70</td>\n",
       "      <td>SP1132</td>\n",
       "      <td>f154cc00-e118-4d4c-b9d2-2f1f71e0a964</td>\n",
       "      <td>SA1580</td>\n",
       "      <td>f154cc00-e118-4d4c-b9d2-2f1f71e0a964</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>2c4bd7a6-721a-4cd1-aa40-90cecb01c32c</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.5c2ab9dd-c1ed-4d57-b68d-56a7b52b80e0.STA...</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>4a81f567-fb9b-4561-b2b7-c099c206a2ce</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.5c2ab9dd-c1ed-4d57-b68d-56a7b52b80e0.Top...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BLCA-US::1e308b12-0590-4dae-94d0-a539fcf25df7</td>\n",
       "      <td>Whitelist</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>1e308b12-0590-4dae-94d0-a539fcf25df7</td>\n",
       "      <td>DO479</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>0.00</td>\n",
       "      <td>59b99c9a-d1c0-4cc0-953b-4358bcf9f392</td>\n",
       "      <td>SP968</td>\n",
       "      <td>d6e91f4c-38c0-4393-9cb7-be076663dff3</td>\n",
       "      <td>SA3784</td>\n",
       "      <td>d6e91f4c-38c0-4393-9cb7-be076663dff3</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>fa36e89d-55b8-4c72-9392-80113efde076</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.c878fe80-0a2e-43d2-ab62-1214fc34c282.bam</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>58624910-e5a1-11e5-8a85-b49cebf72b8f</td>\n",
       "      <td>False</td>\n",
       "      <td>PCAWG.c878fe80-0a2e-43d2-ab62-1214fc34c282_min...</td>\n",
       "      <td>1</td>\n",
       "      <td>e9889071-4c6e-4761-9d65-06c6b5989fb7</td>\n",
       "      <td>SP967</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>SA3742</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>86.05</td>\n",
       "      <td>0.1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>416911eb-e10f-4edd-8f07-5e87b0228a11</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.066b5025-1a70-4db4-bd98-964b8c9939c3.bam</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>58624910-e5a1-11e5-8a85-b49cebf72b8f</td>\n",
       "      <td>False</td>\n",
       "      <td>PCAWG.066b5025-1a70-4db4-bd98-964b8c9939c3_min...</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>67959d40-d34c-11e5-bc3a-f8135c6878de</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>True</td>\n",
       "      <td>838819c6-29d1-4d45-b534-d198dfffdaa9</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>a44f0d88-76af-11e5-b61f-84cb5485c5b1</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>5c58c28e-b8b8-4137-bf8d-526fffab8338</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>True</td>\n",
       "      <td>7e45c6f6-0b5c-474f-b251-2bf58be35290</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>854c515b-bade-4f10-8f24-a46b0abd1507</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>True</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>595384e4-c0e1-47d2-a1e3-c1aa7f9eeea6</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>e9889071-4c6e-4761-9d65-06c6b5989fb7</td>\n",
       "      <td>SP967</td>\n",
       "      <td>429f0e9d-c160-4b36-a31b-dab1097c6129</td>\n",
       "      <td>SA3760</td>\n",
       "      <td>429f0e9d-c160-4b36-a31b-dab1097c6129</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>5e5f155b-0c8c-4917-b47c-606fda32b0e8</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.ba7d499b-049b-4ba2-be02-e513f81d453e.STA...</td>\n",
       "      <td>https://gtrepo-osdc-tcga.annailabs.com/</td>\n",
       "      <td>4c45ff2a-309f-4167-936e-e5b894ce6b79</td>\n",
       "      <td>True</td>\n",
       "      <td>PCAWG.ba7d499b-049b-4ba2-be02-e513f81d453e.Top...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 donor_unique_id wgs_exclusion_white_gray  \\\n",
       "0  BLCA-US::096b4f32-10c1-4737-a0dd-cae04c54ee33                Whitelist   \n",
       "1  BLCA-US::178b28cd-99c3-48dc-8d09-1ef71b4cee80                Whitelist   \n",
       "2  BLCA-US::1e308b12-0590-4dae-94d0-a539fcf25df7                Whitelist   \n",
       "\n",
       "  dcc_project_code                    submitter_donor_id icgc_donor_id  \\\n",
       "0          BLCA-US  096b4f32-10c1-4737-a0dd-cae04c54ee33         DO804   \n",
       "1          BLCA-US  178b28cd-99c3-48dc-8d09-1ef71b4cee80         DO555   \n",
       "2          BLCA-US  1e308b12-0590-4dae-94d0-a539fcf25df7         DO479   \n",
       "\n",
       "   mar2016_donor  santa_cruz_pilot  validation_by_deep_seq   TiN  \\\n",
       "0           True              True                    True  0.01   \n",
       "1           True              True                    True  0.00   \n",
       "2           True              True                    True  0.00   \n",
       "\n",
       "       normal_wgs_submitter_specimen_id normal_wgs_icgc_specimen_id  \\\n",
       "0  27461a27-26eb-4c2c-9c54-e16fbd32c615                      SP1682   \n",
       "1  cd3cfb26-e66f-408e-81f6-3b61c247c976                      SP1135   \n",
       "2  59b99c9a-d1c0-4cc0-953b-4358bcf9f392                       SP968   \n",
       "\n",
       "         normal_wgs_submitter_sample_id normal_wgs_icgc_sample_id  \\\n",
       "0  e0fccaf5-925a-41f9-b87c-cd5ee4aecb59                    SA5237   \n",
       "1  c1da8eed-4919-4ba5-a735-3fba476c18a7                    SA1598   \n",
       "2  d6e91f4c-38c0-4393-9cb7-be076663dff3                    SA3784   \n",
       "\n",
       "                  normal_wgs_aliquot_id  \\\n",
       "0  e0fccaf5-925a-41f9-b87c-cd5ee4aecb59   \n",
       "1  c1da8eed-4919-4ba5-a735-3fba476c18a7   \n",
       "2  d6e91f4c-38c0-4393-9cb7-be076663dff3   \n",
       "\n",
       "        normal_wgs_bwa_alignment_gnos_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "       normal_wgs_bwa_alignment_gnos_id  is_mar2016_normal_wgs_bwa_alignment  \\\n",
       "0  f22a72c5-73c8-478d-b03e-04599b9d5321                                 True   \n",
       "1  964bac8f-43cc-4a47-b233-53798d287029                                 True   \n",
       "2  fa36e89d-55b8-4c72-9392-80113efde076                                 True   \n",
       "\n",
       "           normal_wgs_bwa_alignment_bam_file_name  \\\n",
       "0  PCAWG.67455c36-aa47-4cc4-8b6d-9a9012b616ed.bam   \n",
       "1  PCAWG.13985b6b-d7b6-4efd-9c53-288f4d269a0c.bam   \n",
       "2  PCAWG.c878fe80-0a2e-43d2-ab62-1214fc34c282.bam   \n",
       "\n",
       "              normal_wgs_minibam_gnos_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "             normal_wgs_minibam_gnos_id  is_mar2016_normal_wgs_minibam  \\\n",
       "0  7e442040-f6ea-11e5-bd75-997f5888a0e8                          False   \n",
       "1  0b6a5d16-e617-11e5-ad33-d98536d081ee                          False   \n",
       "2  58624910-e5a1-11e5-8a85-b49cebf72b8f                          False   \n",
       "\n",
       "                    normal_wgs_minibam_bam_file_name  \\\n",
       "0  PCAWG.67455c36-aa47-4cc4-8b6d-9a9012b616ed_min...   \n",
       "1  PCAWG.13985b6b-d7b6-4efd-9c53-288f4d269a0c_min...   \n",
       "2  PCAWG.c878fe80-0a2e-43d2-ab62-1214fc34c282_min...   \n",
       "\n",
       "   tumor_wgs_specimen_count       tumor_wgs_submitter_specimen_id  \\\n",
       "0                         1  52f538ef-b05d-4c76-9976-ce6d49158016   \n",
       "1                         1  59d6683f-5eb7-493d-8e8e-78b88be2cd70   \n",
       "2                         1  e9889071-4c6e-4761-9d65-06c6b5989fb7   \n",
       "\n",
       "  tumor_wgs_icgc_specimen_id         tumor_wgs_submitter_sample_id  \\\n",
       "0                     SP1677  301d6ce3-4099-4c1d-8e50-c04b7ce91450   \n",
       "1                     SP1132  4838b5a9-968c-4178-bffb-3fafe1f6dc09   \n",
       "2                      SP967  c66c92d5-df65-46e6-861d-d8a98808e6a3   \n",
       "\n",
       "  tumor_wgs_icgc_sample_id                  tumor_wgs_aliquot_id  \\\n",
       "0                   SA5195  301d6ce3-4099-4c1d-8e50-c04b7ce91450   \n",
       "1                   SA1556  4838b5a9-968c-4178-bffb-3fafe1f6dc09   \n",
       "2                   SA3742  c66c92d5-df65-46e6-861d-d8a98808e6a3   \n",
       "\n",
       "  tumor_wgs_oxog_score tumor_wgs_ContEST tumor_wgs_Stars  \\\n",
       "0                87.52               0.1             5.0   \n",
       "1                54.55               0.1             5.0   \n",
       "2                86.05               0.1             5.0   \n",
       "\n",
       "         tumor_wgs_bwa_alignment_gnos_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "        tumor_wgs_bwa_alignment_gnos_id is_mar2016_tumor_wgs_bwa_alignment  \\\n",
       "0  a6e8dd23-c8a5-445a-ae4b-b9f92ed6a73e                               True   \n",
       "1  437e11a0-4137-4614-9f64-c5e798c8bb33                               True   \n",
       "2  416911eb-e10f-4edd-8f07-5e87b0228a11                               True   \n",
       "\n",
       "            tumor_wgs_bwa_alignment_bam_file_name  \\\n",
       "0  PCAWG.a833602e-7ee7-442b-9bf9-f6489cd64ba2.bam   \n",
       "1  PCAWG.af830047-822a-4dcc-88bc-68f7440a208d.bam   \n",
       "2  PCAWG.066b5025-1a70-4db4-bd98-964b8c9939c3.bam   \n",
       "\n",
       "               tumor_wgs_minibam_gnos_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "              tumor_wgs_minibam_gnos_id is_mar2016_tumor_wgs_minibam  \\\n",
       "0  7e442040-f6ea-11e5-bd75-997f5888a0e8                        False   \n",
       "1  0b6a5d16-e617-11e5-ad33-d98536d081ee                        False   \n",
       "2  58624910-e5a1-11e5-8a85-b49cebf72b8f                        False   \n",
       "\n",
       "                     tumor_wgs_minibam_bam_file_name  \\\n",
       "0  PCAWG.a833602e-7ee7-442b-9bf9-f6489cd64ba2_min...   \n",
       "1  PCAWG.af830047-822a-4dcc-88bc-68f7440a208d_min...   \n",
       "2  PCAWG.066b5025-1a70-4db4-bd98-964b8c9939c3_min...   \n",
       "\n",
       "               sanger_variant_calling_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "         sanger_variant_calling_gnos_id  \\\n",
       "0  8935fe36-ceec-11e5-bedc-767d4d100685   \n",
       "1  714e7b6a-d34a-11e5-90f0-f8135c6878de   \n",
       "2  67959d40-d34c-11e5-bc3a-f8135c6878de   \n",
       "\n",
       "  sanger_variant_calling_file_name_prefix  is_mar2016_sanger_variant_calling  \\\n",
       "0    301d6ce3-4099-4c1d-8e50-c04b7ce91450                               True   \n",
       "1    4838b5a9-968c-4178-bffb-3fafe1f6dc09                               True   \n",
       "2    c66c92d5-df65-46e6-861d-d8a98808e6a3                               True   \n",
       "\n",
       "  sanger_variant_calling_deprecated_gnos_id  \\\n",
       "0      5230346f-987e-44c3-b9fe-dddb5829e247   \n",
       "1      c0387aa4-9937-45f7-bf3a-e2da65457738   \n",
       "2      838819c6-29d1-4d45-b534-d198dfffdaa9   \n",
       "\n",
       "            dkfz_embl_variant_calling_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "      dkfz_embl_variant_calling_gnos_id  \\\n",
       "0  4269588d-5012-4eee-ab9a-9c48897dae69   \n",
       "1  875fdb54-76ae-11e5-abee-bff16f24f4a1   \n",
       "2  a44f0d88-76af-11e5-b61f-84cb5485c5b1   \n",
       "\n",
       "  dkfz_embl_variant_calling_file_name_prefix  \\\n",
       "0       301d6ce3-4099-4c1d-8e50-c04b7ce91450   \n",
       "1       4838b5a9-968c-4178-bffb-3fafe1f6dc09   \n",
       "2       c66c92d5-df65-46e6-861d-d8a98808e6a3   \n",
       "\n",
       "   is_mar2016_dkfz_embl_variant_calling  \\\n",
       "0                                  True   \n",
       "1                                  True   \n",
       "2                                  True   \n",
       "\n",
       "  dkfz_embl_variant_calling_deprecated_gnos_id  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "\n",
       "                broad_variant_calling_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "          broad_variant_calling_gnos_id  \\\n",
       "0  ac34e35d-21b1-4b11-b42d-155cba1a547f   \n",
       "1  d989e42d-354c-4652-9921-27f79ba31c2e   \n",
       "2  5c58c28e-b8b8-4137-bf8d-526fffab8338   \n",
       "\n",
       "  broad_variant_calling_file_name_prefix  is_mar2016_broad_variant_calling  \\\n",
       "0   301d6ce3-4099-4c1d-8e50-c04b7ce91450                              True   \n",
       "1   4838b5a9-968c-4178-bffb-3fafe1f6dc09                              True   \n",
       "2   c66c92d5-df65-46e6-861d-d8a98808e6a3                              True   \n",
       "\n",
       "  broad_variant_calling_deprecated_gnos_id  \\\n",
       "0     d12904ff-4ce6-4d7b-a0d7-69e18a3822cc   \n",
       "1     f3c6d06b-09ff-4e0d-821d-80bd2b04106f   \n",
       "2     7e45c6f6-0b5c-474f-b251-2bf58be35290   \n",
       "\n",
       "                 muse_variant_calling_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "           muse_variant_calling_gnos_id muse_variant_calling_file_name_prefix  \\\n",
       "0  48077833-99ed-49c6-88a1-80cd2e50755a  301d6ce3-4099-4c1d-8e50-c04b7ce91450   \n",
       "1  d9ec79b7-0ccb-4312-b84d-b90200eb77bf  4838b5a9-968c-4178-bffb-3fafe1f6dc09   \n",
       "2  854c515b-bade-4f10-8f24-a46b0abd1507  c66c92d5-df65-46e6-861d-d8a98808e6a3   \n",
       "\n",
       "   is_mar2016_muse_variant_calling           broad_tar_variant_calling_repo  \\\n",
       "0                             True  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1                             True  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2                             True  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "      broad_tar_variant_calling_gnos_id  \\\n",
       "0  38ce6165-c324-4642-89cf-984d738ee08a   \n",
       "1  9f6910ec-80dc-4270-b415-26c00f5ad281   \n",
       "2  595384e4-c0e1-47d2-a1e3-c1aa7f9eeea6   \n",
       "\n",
       "  broad_tar_variant_calling_file_name_prefix  \\\n",
       "0       301d6ce3-4099-4c1d-8e50-c04b7ce91450   \n",
       "1       4838b5a9-968c-4178-bffb-3fafe1f6dc09   \n",
       "2       c66c92d5-df65-46e6-861d-d8a98808e6a3   \n",
       "\n",
       "   is_mar2016_broad_tar_variant_calling  normal_wgs_has_matched_rna_seq  \\\n",
       "0                                  True                           False   \n",
       "1                                  True                           False   \n",
       "2                                  True                           False   \n",
       "\n",
       "  tumor_wgs_has_matched_rna_seq normal_rna_seq_submitter_specimen_id  \\\n",
       "0                          True                                  NaN   \n",
       "1                          True                                  NaN   \n",
       "2                          True                                  NaN   \n",
       "\n",
       "  normal_rna_seq_icgc_specimen_id normal_rna_seq_submitter_sample_id  \\\n",
       "0                             NaN                                NaN   \n",
       "1                             NaN                                NaN   \n",
       "2                             NaN                                NaN   \n",
       "\n",
       "  normal_rna_seq_icgc_sample_id normal_rna_seq_aliquot_id  \\\n",
       "0                           NaN                       NaN   \n",
       "1                           NaN                       NaN   \n",
       "2                           NaN                       NaN   \n",
       "\n",
       "  normal_rna_seq_star_alignment_gnos_repo  \\\n",
       "0                                     NaN   \n",
       "1                                     NaN   \n",
       "2                                     NaN   \n",
       "\n",
       "  normal_rna_seq_star_alignment_gnos_id  \\\n",
       "0                                   NaN   \n",
       "1                                   NaN   \n",
       "2                                   NaN   \n",
       "\n",
       "  is_mar2016_normal_rna_seq_star_alignment  \\\n",
       "0                                      NaN   \n",
       "1                                      NaN   \n",
       "2                                      NaN   \n",
       "\n",
       "  normal_rna_seq_star_alignment_bam_file_name  \\\n",
       "0                                         NaN   \n",
       "1                                         NaN   \n",
       "2                                         NaN   \n",
       "\n",
       "  normal_rna_seq_tophat_alignment_gnos_repo  \\\n",
       "0                                       NaN   \n",
       "1                                       NaN   \n",
       "2                                       NaN   \n",
       "\n",
       "  normal_rna_seq_tophat_alignment_gnos_id  \\\n",
       "0                                     NaN   \n",
       "1                                     NaN   \n",
       "2                                     NaN   \n",
       "\n",
       "  is_mar2016_normal_rna_seq_tophat_alignment  \\\n",
       "0                                        NaN   \n",
       "1                                        NaN   \n",
       "2                                        NaN   \n",
       "\n",
       "  normal_rna_seq_tophat_alignment_bam_file_name  tumor_rna_seq_specimen_count  \\\n",
       "0                                           NaN                           1.0   \n",
       "1                                           NaN                           1.0   \n",
       "2                                           NaN                           1.0   \n",
       "\n",
       "    tumor_rna_seq_submitter_specimen_id tumor_rna_seq_icgc_specimen_id  \\\n",
       "0  52f538ef-b05d-4c76-9976-ce6d49158016                         SP1677   \n",
       "1  59d6683f-5eb7-493d-8e8e-78b88be2cd70                         SP1132   \n",
       "2  e9889071-4c6e-4761-9d65-06c6b5989fb7                          SP967   \n",
       "\n",
       "      tumor_rna_seq_submitter_sample_id tumor_rna_seq_icgc_sample_id  \\\n",
       "0  22e154de-0e3b-443b-8420-48d68d6c1ce4                       SA5213   \n",
       "1  f154cc00-e118-4d4c-b9d2-2f1f71e0a964                       SA1580   \n",
       "2  429f0e9d-c160-4b36-a31b-dab1097c6129                       SA3760   \n",
       "\n",
       "               tumor_rna_seq_aliquot_id  \\\n",
       "0  22e154de-0e3b-443b-8420-48d68d6c1ce4   \n",
       "1  f154cc00-e118-4d4c-b9d2-2f1f71e0a964   \n",
       "2  429f0e9d-c160-4b36-a31b-dab1097c6129   \n",
       "\n",
       "    tumor_rna_seq_star_alignment_gnos_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "   tumor_rna_seq_star_alignment_gnos_id  \\\n",
       "0  a63c9c0a-c8aa-44dc-8915-08f55109a5c9   \n",
       "1  2c4bd7a6-721a-4cd1-aa40-90cecb01c32c   \n",
       "2  5e5f155b-0c8c-4917-b47c-606fda32b0e8   \n",
       "\n",
       "  is_mar2016_tumor_rna_seq_star_alignment  \\\n",
       "0                                    True   \n",
       "1                                    True   \n",
       "2                                    True   \n",
       "\n",
       "          tumor_rna_seq_star_alignment_bam_file_name  \\\n",
       "0  PCAWG.f4288632-a94f-4777-a5d6-1a76b12abcc5.STA...   \n",
       "1  PCAWG.5c2ab9dd-c1ed-4d57-b68d-56a7b52b80e0.STA...   \n",
       "2  PCAWG.ba7d499b-049b-4ba2-be02-e513f81d453e.STA...   \n",
       "\n",
       "  tumor_rna_seq_tophat_alignment_gnos_repo  \\\n",
       "0  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "1  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "2  https://gtrepo-osdc-tcga.annailabs.com/   \n",
       "\n",
       "  tumor_rna_seq_tophat_alignment_gnos_id  \\\n",
       "0   81438f7c-ee87-44a7-acd1-562ec8f386f0   \n",
       "1   4a81f567-fb9b-4561-b2b7-c099c206a2ce   \n",
       "2   4c45ff2a-309f-4167-936e-e5b894ce6b79   \n",
       "\n",
       "  is_mar2016_tumor_rna_seq_tophat_alignment  \\\n",
       "0                                      True   \n",
       "1                                      True   \n",
       "2                                      True   \n",
       "\n",
       "        tumor_rna_seq_tophat_alignment_bam_file_name  \n",
       "0  PCAWG.f4288632-a94f-4777-a5d6-1a76b12abcc5.Top...  \n",
       "1  PCAWG.5c2ab9dd-c1ed-4d57-b68d-56a7b52b80e0.Top...  \n",
       "2  PCAWG.ba7d499b-049b-4ba2-be02-e513f81d453e.Top...  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "release_may2016_table.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "0d281bb4-f551-4c90-8895-0043c0049d7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#release_may2016_table_small = release_may2016_table[['submitter_donor_id', 'normal_wgs_submitter_specimen_id','normal_wgs_icgc_specimen_id', 'normal_wgs_submitter_sample_id', 'normal_wgs_icgc_sample_id', 'tumor_wgs_icgc_specimen_id','tumor_wgs_submitter_sample_id',\t'tumor_wgs_icgc_sample_id', 'tumor_rna_seq_icgc_specimen_id', 'tumor_rna_seq_icgc_sample_id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "b7ba7f43-0d0e-41b9-8874-b784d1d6327d",
   "metadata": {},
   "outputs": [],
   "source": [
    "release_may2016_table_small = release_may2016_table[['normal_wgs_submitter_specimen_id','normal_wgs_icgc_specimen_id', 'normal_wgs_submitter_sample_id', 'normal_wgs_icgc_sample_id', 'tumor_wgs_icgc_specimen_id','tumor_wgs_submitter_sample_id',\t'tumor_wgs_icgc_sample_id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "e9978c19-512b-4291-b1a7-470d94daf2fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>normal_wgs_submitter_specimen_id</th>\n",
       "      <th>normal_wgs_icgc_specimen_id</th>\n",
       "      <th>normal_wgs_submitter_sample_id</th>\n",
       "      <th>normal_wgs_icgc_sample_id</th>\n",
       "      <th>tumor_wgs_icgc_specimen_id</th>\n",
       "      <th>tumor_wgs_submitter_sample_id</th>\n",
       "      <th>tumor_wgs_icgc_sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27461a27-26eb-4c2c-9c54-e16fbd32c615</td>\n",
       "      <td>SP1682</td>\n",
       "      <td>e0fccaf5-925a-41f9-b87c-cd5ee4aecb59</td>\n",
       "      <td>SA5237</td>\n",
       "      <td>SP1677</td>\n",
       "      <td>301d6ce3-4099-4c1d-8e50-c04b7ce91450</td>\n",
       "      <td>SA5195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cd3cfb26-e66f-408e-81f6-3b61c247c976</td>\n",
       "      <td>SP1135</td>\n",
       "      <td>c1da8eed-4919-4ba5-a735-3fba476c18a7</td>\n",
       "      <td>SA1598</td>\n",
       "      <td>SP1132</td>\n",
       "      <td>4838b5a9-968c-4178-bffb-3fafe1f6dc09</td>\n",
       "      <td>SA1556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>59b99c9a-d1c0-4cc0-953b-4358bcf9f392</td>\n",
       "      <td>SP968</td>\n",
       "      <td>d6e91f4c-38c0-4393-9cb7-be076663dff3</td>\n",
       "      <td>SA3784</td>\n",
       "      <td>SP967</td>\n",
       "      <td>c66c92d5-df65-46e6-861d-d8a98808e6a3</td>\n",
       "      <td>SA3742</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       normal_wgs_submitter_specimen_id normal_wgs_icgc_specimen_id  \\\n",
       "0  27461a27-26eb-4c2c-9c54-e16fbd32c615                      SP1682   \n",
       "1  cd3cfb26-e66f-408e-81f6-3b61c247c976                      SP1135   \n",
       "2  59b99c9a-d1c0-4cc0-953b-4358bcf9f392                       SP968   \n",
       "\n",
       "         normal_wgs_submitter_sample_id normal_wgs_icgc_sample_id  \\\n",
       "0  e0fccaf5-925a-41f9-b87c-cd5ee4aecb59                    SA5237   \n",
       "1  c1da8eed-4919-4ba5-a735-3fba476c18a7                    SA1598   \n",
       "2  d6e91f4c-38c0-4393-9cb7-be076663dff3                    SA3784   \n",
       "\n",
       "  tumor_wgs_icgc_specimen_id         tumor_wgs_submitter_sample_id  \\\n",
       "0                     SP1677  301d6ce3-4099-4c1d-8e50-c04b7ce91450   \n",
       "1                     SP1132  4838b5a9-968c-4178-bffb-3fafe1f6dc09   \n",
       "2                      SP967  c66c92d5-df65-46e6-861d-d8a98808e6a3   \n",
       "\n",
       "  tumor_wgs_icgc_sample_id  \n",
       "0                   SA5195  \n",
       "1                   SA1556  \n",
       "2                   SA3742  "
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "release_may2016_table_small.head(3)\n",
    "# normal and tumour sample/specimen ids are mutually exclusive, we work only with the tumour data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "17cdeb62-0111-4eac-9dc0-4f5590989566",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2778, 7)"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_annotation_table = annotations_table_small.copy()  # Make a copy to avoid modifying the original DataFrame\n",
    "combined_annotation_table['matched_tumor_wgs_icgc_specimen_id'] = None\n",
    "#combined_annotation_table['matched_tumor_wgs_icgc_sample_id'] = None\n",
    "#combined_annotation_table['matched_tumor_wgs_submitter_sample_id'] = None\n",
    "\n",
    "# Iterate over each icgc_sample_id in combined_annotation_table\n",
    "for index, sample_id in combined_annotation_table['icgc_sample_id'].items():\n",
    "    # Check if the sample_id matches any tumor_wgs_icgc_sample_id in release_may2016_table_small\n",
    "    match = release_may2016_table_small[release_may2016_table_small['tumor_wgs_icgc_sample_id'] == sample_id]\n",
    "    if not match.empty:\n",
    "        # Add the matching tumor_wgs_icgc_sample_id, tumor_wgs_icgc_specimen_id, and tumor_wgs_submitter_sample_id to the new columns\n",
    "        combined_annotation_table.loc[index, 'matched_tumor_wgs_icgc_specimen_id'] = match['tumor_wgs_icgc_specimen_id'].values[0]\n",
    "        #combined_annotation_table.loc[index, 'matched_tumor_wgs_icgc_sample_id'] = match['tumor_wgs_icgc_sample_id'].values[0]\n",
    "        #combined_annotation_table.loc[index, 'matched_tumor_wgs_submitter_sample_id'] = match['tumor_wgs_submitter_sample_id'].values[0]\n",
    "\n",
    "#Reorder columns\n",
    "combined_annotation_table=combined_annotation_table[['icgc_sample_id', 'matched_tumor_wgs_icgc_specimen_id', 'samplename','icgc_donor_id','purity', 'histology_abbreviation','cancer_type']] \n",
    "combined_annotation_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "db52007f-dd01-439b-a800-c56e6b5bb1a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_tumor_wgs_icgc_specimen_id</th>\n",
       "      <th>samplename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>purity</th>\n",
       "      <th>histology_abbreviation</th>\n",
       "      <th>cancer_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SA505245</td>\n",
       "      <td>SP101724</td>\n",
       "      <td>0009b464-b376-4fbc-8a56-da538269a02f</td>\n",
       "      <td>DO46416</td>\n",
       "      <td>0.885</td>\n",
       "      <td>Ovary-AdenoCA</td>\n",
       "      <td>OV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SA413814</td>\n",
       "      <td>SP79365</td>\n",
       "      <td>003819bc-c415-4e76-887c-931d60ed39e7</td>\n",
       "      <td>DO36062</td>\n",
       "      <td>0.774</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>PBCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SA501385</td>\n",
       "      <td>SP98853</td>\n",
       "      <td>0040b1b6-b07a-4b6e-90ef-133523eaf412</td>\n",
       "      <td>DO45049</td>\n",
       "      <td>0.800</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>LINC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_sample_id matched_tumor_wgs_icgc_specimen_id  \\\n",
       "0       SA505245                           SP101724   \n",
       "1       SA413814                            SP79365   \n",
       "2       SA501385                            SP98853   \n",
       "\n",
       "                             samplename icgc_donor_id  purity  \\\n",
       "0  0009b464-b376-4fbc-8a56-da538269a02f       DO46416   0.885   \n",
       "1  003819bc-c415-4e76-887c-931d60ed39e7       DO36062   0.774   \n",
       "2  0040b1b6-b07a-4b6e-90ef-133523eaf412       DO45049   0.800   \n",
       "\n",
       "  histology_abbreviation cancer_type  \n",
       "0          Ovary-AdenoCA          OV  \n",
       "1          CNS-PiloAstro        PBCA  \n",
       "2              Liver-HCC        LINC  "
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_annotation_table.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "17bd522c-b9c5-4e79-99bf-509adc4e201e",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/omics/groups/OE0606/internal/jonas/genomic_data/PCAWG_sigProfiler_SBS_signatures_in_samples.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[82], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m###\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m### Now for PCAWG_sigProfiler_SBS_signatures_in_samples.csv table ###\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m SBS_signatures_in_samples_table \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m/omics/groups/OE0606/internal/jonas/genomic_data/PCAWG_sigProfiler_SBS_signatures_in_samples.csv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(SBS_signatures_in_samples_table\u001b[38;5;241m.\u001b[39mshape)\n",
      "File \u001b[0;32m/omics/groups/OE0606/internal/jonas/miniconda/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1026\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m   1013\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m   1014\u001b[0m     dialect,\n\u001b[1;32m   1015\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1022\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[1;32m   1023\u001b[0m )\n\u001b[1;32m   1024\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m-> 1026\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/omics/groups/OE0606/internal/jonas/miniconda/lib/python3.12/site-packages/pandas/io/parsers/readers.py:620\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    617\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    619\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 620\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    622\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    623\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m/omics/groups/OE0606/internal/jonas/miniconda/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1620\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1617\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1619\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1620\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/omics/groups/OE0606/internal/jonas/miniconda/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1880\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1878\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1879\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1880\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1881\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1882\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1883\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1884\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1885\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1886\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1887\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1888\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1889\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1890\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1891\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m/omics/groups/OE0606/internal/jonas/miniconda/lib/python3.12/site-packages/pandas/io/common.py:873\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    868\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    869\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    870\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    871\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    872\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 873\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    874\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    875\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    876\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    877\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    878\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    879\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    880\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    881\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    882\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/omics/groups/OE0606/internal/jonas/genomic_data/PCAWG_sigProfiler_SBS_signatures_in_samples.csv'"
     ]
    }
   ],
   "source": [
    "###\n",
    "### Now for PCAWG_sigProfiler_SBS_signatures_in_samples.csv table ###\n",
    "\n",
    "SBS_signatures_in_samples_table = pd.read_csv('/omics/groups/OE0606/internal/jonas/genomic_data/PCAWG_sigProfiler_SBS_signatures_in_samples.csv')\n",
    "print(SBS_signatures_in_samples_table.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "1b2a011e-25c5-45ee-9b80-007203affcef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer Types</th>\n",
       "      <th>Sample Names</th>\n",
       "      <th>Accuracy</th>\n",
       "      <th>SBS1</th>\n",
       "      <th>SBS2</th>\n",
       "      <th>SBS3</th>\n",
       "      <th>SBS4</th>\n",
       "      <th>SBS5</th>\n",
       "      <th>SBS6</th>\n",
       "      <th>SBS7a</th>\n",
       "      <th>SBS7b</th>\n",
       "      <th>SBS7c</th>\n",
       "      <th>SBS7d</th>\n",
       "      <th>SBS8</th>\n",
       "      <th>SBS9</th>\n",
       "      <th>SBS10a</th>\n",
       "      <th>SBS10b</th>\n",
       "      <th>SBS11</th>\n",
       "      <th>SBS12</th>\n",
       "      <th>SBS13</th>\n",
       "      <th>SBS14</th>\n",
       "      <th>SBS15</th>\n",
       "      <th>SBS16</th>\n",
       "      <th>SBS17a</th>\n",
       "      <th>SBS17b</th>\n",
       "      <th>SBS18</th>\n",
       "      <th>SBS19</th>\n",
       "      <th>SBS20</th>\n",
       "      <th>SBS21</th>\n",
       "      <th>SBS22</th>\n",
       "      <th>SBS23</th>\n",
       "      <th>SBS24</th>\n",
       "      <th>SBS25</th>\n",
       "      <th>SBS26</th>\n",
       "      <th>SBS27</th>\n",
       "      <th>SBS28</th>\n",
       "      <th>SBS29</th>\n",
       "      <th>SBS30</th>\n",
       "      <th>SBS31</th>\n",
       "      <th>SBS32</th>\n",
       "      <th>SBS33</th>\n",
       "      <th>SBS34</th>\n",
       "      <th>SBS35</th>\n",
       "      <th>SBS36</th>\n",
       "      <th>SBS37</th>\n",
       "      <th>SBS38</th>\n",
       "      <th>SBS39</th>\n",
       "      <th>SBS40</th>\n",
       "      <th>SBS41</th>\n",
       "      <th>SBS42</th>\n",
       "      <th>SBS43</th>\n",
       "      <th>SBS44</th>\n",
       "      <th>SBS45</th>\n",
       "      <th>SBS46</th>\n",
       "      <th>SBS47</th>\n",
       "      <th>SBS48</th>\n",
       "      <th>SBS49</th>\n",
       "      <th>SBS50</th>\n",
       "      <th>SBS51</th>\n",
       "      <th>SBS52</th>\n",
       "      <th>SBS53</th>\n",
       "      <th>SBS54</th>\n",
       "      <th>SBS55</th>\n",
       "      <th>SBS56</th>\n",
       "      <th>SBS57</th>\n",
       "      <th>SBS58</th>\n",
       "      <th>SBS59</th>\n",
       "      <th>SBS60</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Biliary-AdenoCA</td>\n",
       "      <td>SP117655</td>\n",
       "      <td>0.968</td>\n",
       "      <td>1496</td>\n",
       "      <td>1296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1825</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>806</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9477</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Biliary-AdenoCA</td>\n",
       "      <td>SP117556</td>\n",
       "      <td>0.963</td>\n",
       "      <td>985</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>922</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3181</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Biliary-AdenoCA</td>\n",
       "      <td>SP117627</td>\n",
       "      <td>0.973</td>\n",
       "      <td>1110</td>\n",
       "      <td>528</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1453</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2805</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Cancer Types Sample Names  Accuracy  SBS1  SBS2  SBS3  SBS4  SBS5  SBS6  \\\n",
       "0  Biliary-AdenoCA     SP117655     0.968  1496  1296     0     0  1825     0   \n",
       "1  Biliary-AdenoCA     SP117556     0.963   985     0     0     0   922     0   \n",
       "2  Biliary-AdenoCA     SP117627     0.973  1110   528     0     0  1453     0   \n",
       "\n",
       "   SBS7a  SBS7b  SBS7c  SBS7d  SBS8  SBS9  SBS10a  SBS10b  SBS11  SBS12  \\\n",
       "0      0      0      0      0     0     0       0       0      0      0   \n",
       "1      0      0      0      0     0     0       0       0      0      0   \n",
       "2      0      0      0      0     0     0       0       0      0      0   \n",
       "\n",
       "   SBS13  SBS14  SBS15  SBS16  SBS17a  SBS17b  SBS18  SBS19  SBS20  SBS21  \\\n",
       "0    806      0      0      0       0       0      0      0      0      0   \n",
       "1      0      0      0      0       0       0      0      0      0      0   \n",
       "2     27      0      0      0       0       0      0      0      0      0   \n",
       "\n",
       "   SBS22  SBS23  SBS24  SBS25  SBS26  SBS27  SBS28  SBS29  SBS30  SBS31  \\\n",
       "0      0      0      0      0      0      0      0      0      0      0   \n",
       "1      0      0      0      0      0      0      0      0      0      0   \n",
       "2      0      0      0      0      0      0      0      0      0      0   \n",
       "\n",
       "   SBS32  SBS33  SBS34  SBS35  SBS36  SBS37  SBS38  SBS39  SBS40  SBS41  \\\n",
       "0      0      0      0      0      0      0      0      0   9477      0   \n",
       "1      0      0      0      0      0      0      0      0   3181      0   \n",
       "2      0      0      0      0      0      0      0      0   2805      0   \n",
       "\n",
       "   SBS42  SBS43  SBS44  SBS45  SBS46  SBS47  SBS48  SBS49  SBS50  SBS51  \\\n",
       "0      0      0      0      0      0      0      0      0      0      0   \n",
       "1      0      0      0      0      0      0      0      0      0      0   \n",
       "2      0      0      0      0      0      0      0      0      0      0   \n",
       "\n",
       "   SBS52  SBS53  SBS54  SBS55  SBS56  SBS57  SBS58  SBS59  SBS60  \n",
       "0      0      0      0      0      0      0      0      0      0  \n",
       "1      0      0      0      0      0      0      0      0      0  \n",
       "2      0      0      0      0      0      0      0      0      0  "
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SBS_signatures_in_samples_table.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "bd813db5-70d5-4f30-be24-cdfa6783a76b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2778, 75)"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_annotations_SBS_signatures = pd.merge(\n",
    "    combined_annotation_table,\n",
    "    SBS_signatures_in_samples_table,\n",
    "    left_on='matched_tumor_wgs_icgc_specimen_id',\n",
    "    right_on='Sample Names',\n",
    "    how='left'\n",
    ")\n",
    "combined_annotations_SBS_signatures.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "d868c6fd-9843-4d3d-a0aa-6394e8df2b88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_tumor_wgs_icgc_specimen_id</th>\n",
       "      <th>samplename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>purity</th>\n",
       "      <th>histology_abbreviation</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>Cancer Types</th>\n",
       "      <th>Sample Names</th>\n",
       "      <th>Accuracy</th>\n",
       "      <th>SBS1</th>\n",
       "      <th>SBS2</th>\n",
       "      <th>SBS3</th>\n",
       "      <th>SBS4</th>\n",
       "      <th>SBS5</th>\n",
       "      <th>SBS6</th>\n",
       "      <th>SBS7a</th>\n",
       "      <th>SBS7b</th>\n",
       "      <th>SBS7c</th>\n",
       "      <th>SBS7d</th>\n",
       "      <th>SBS8</th>\n",
       "      <th>SBS9</th>\n",
       "      <th>SBS10a</th>\n",
       "      <th>SBS10b</th>\n",
       "      <th>SBS11</th>\n",
       "      <th>SBS12</th>\n",
       "      <th>SBS13</th>\n",
       "      <th>SBS14</th>\n",
       "      <th>SBS15</th>\n",
       "      <th>SBS16</th>\n",
       "      <th>SBS17a</th>\n",
       "      <th>SBS17b</th>\n",
       "      <th>SBS18</th>\n",
       "      <th>SBS19</th>\n",
       "      <th>SBS20</th>\n",
       "      <th>SBS21</th>\n",
       "      <th>SBS22</th>\n",
       "      <th>SBS23</th>\n",
       "      <th>SBS24</th>\n",
       "      <th>SBS25</th>\n",
       "      <th>SBS26</th>\n",
       "      <th>SBS27</th>\n",
       "      <th>SBS28</th>\n",
       "      <th>SBS29</th>\n",
       "      <th>SBS30</th>\n",
       "      <th>SBS31</th>\n",
       "      <th>SBS32</th>\n",
       "      <th>SBS33</th>\n",
       "      <th>SBS34</th>\n",
       "      <th>SBS35</th>\n",
       "      <th>SBS36</th>\n",
       "      <th>SBS37</th>\n",
       "      <th>SBS38</th>\n",
       "      <th>SBS39</th>\n",
       "      <th>SBS40</th>\n",
       "      <th>SBS41</th>\n",
       "      <th>SBS42</th>\n",
       "      <th>SBS43</th>\n",
       "      <th>SBS44</th>\n",
       "      <th>SBS45</th>\n",
       "      <th>SBS46</th>\n",
       "      <th>SBS47</th>\n",
       "      <th>SBS48</th>\n",
       "      <th>SBS49</th>\n",
       "      <th>SBS50</th>\n",
       "      <th>SBS51</th>\n",
       "      <th>SBS52</th>\n",
       "      <th>SBS53</th>\n",
       "      <th>SBS54</th>\n",
       "      <th>SBS55</th>\n",
       "      <th>SBS56</th>\n",
       "      <th>SBS57</th>\n",
       "      <th>SBS58</th>\n",
       "      <th>SBS59</th>\n",
       "      <th>SBS60</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SA505245</td>\n",
       "      <td>SP101724</td>\n",
       "      <td>0009b464-b376-4fbc-8a56-da538269a02f</td>\n",
       "      <td>DO46416</td>\n",
       "      <td>0.885</td>\n",
       "      <td>Ovary-AdenoCA</td>\n",
       "      <td>OV</td>\n",
       "      <td>Ovary-AdenoCA</td>\n",
       "      <td>SP101724</td>\n",
       "      <td>0.943</td>\n",
       "      <td>792.0</td>\n",
       "      <td>668.0</td>\n",
       "      <td>6361.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>543.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6909.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SA413814</td>\n",
       "      <td>SP79365</td>\n",
       "      <td>003819bc-c415-4e76-887c-931d60ed39e7</td>\n",
       "      <td>DO36062</td>\n",
       "      <td>0.774</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>PBCA</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>SP79365</td>\n",
       "      <td>0.810</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SA501385</td>\n",
       "      <td>SP98853</td>\n",
       "      <td>0040b1b6-b07a-4b6e-90ef-133523eaf412</td>\n",
       "      <td>DO45049</td>\n",
       "      <td>0.800</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>LINC</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>SP98853</td>\n",
       "      <td>0.989</td>\n",
       "      <td>326.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2142.0</td>\n",
       "      <td>6887.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5401.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_sample_id matched_tumor_wgs_icgc_specimen_id  \\\n",
       "0       SA505245                           SP101724   \n",
       "1       SA413814                            SP79365   \n",
       "2       SA501385                            SP98853   \n",
       "\n",
       "                             samplename icgc_donor_id  purity  \\\n",
       "0  0009b464-b376-4fbc-8a56-da538269a02f       DO46416   0.885   \n",
       "1  003819bc-c415-4e76-887c-931d60ed39e7       DO36062   0.774   \n",
       "2  0040b1b6-b07a-4b6e-90ef-133523eaf412       DO45049   0.800   \n",
       "\n",
       "  histology_abbreviation cancer_type   Cancer Types Sample Names  Accuracy  \\\n",
       "0          Ovary-AdenoCA          OV  Ovary-AdenoCA     SP101724     0.943   \n",
       "1          CNS-PiloAstro        PBCA  CNS-PiloAstro      SP79365     0.810   \n",
       "2              Liver-HCC        LINC      Liver-HCC      SP98853     0.989   \n",
       "\n",
       "    SBS1   SBS2    SBS3    SBS4    SBS5  SBS6  SBS7a  SBS7b  SBS7c  SBS7d  \\\n",
       "0  792.0  668.0  6361.0     0.0     0.0   0.0    0.0    0.0    0.0    0.0   \n",
       "1   17.0    0.0     0.0     0.0    87.0   0.0    0.0    0.0    0.0    0.0   \n",
       "2  326.0    0.0     0.0  2142.0  6887.0   0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS8  SBS9  SBS10a  SBS10b  SBS11  SBS12  SBS13  SBS14  SBS15   SBS16  \\\n",
       "0   0.0   0.0     0.0     0.0    0.0    0.0  543.0    0.0    0.0     0.0   \n",
       "1   0.0   0.0     0.0     0.0    0.0    0.0    0.0    0.0    0.0     0.0   \n",
       "2   0.0   0.0     0.0     0.0    0.0    0.0    0.0    0.0    0.0  5401.0   \n",
       "\n",
       "   SBS17a  SBS17b  SBS18  SBS19  SBS20  SBS21  SBS22  SBS23  SBS24  SBS25  \\\n",
       "0     0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1     0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2     0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS26  SBS27  SBS28  SBS29  SBS30  SBS31  SBS32  SBS33  SBS34  SBS35  \\\n",
       "0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS36  SBS37  SBS38  SBS39   SBS40  SBS41  SBS42  SBS43  SBS44  SBS45  \\\n",
       "0    0.0    0.0    0.0    0.0  6909.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1    0.0    0.0    0.0    0.0     0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2    0.0    0.0    0.0    0.0     0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS46  SBS47  SBS48  SBS49  SBS50  SBS51  SBS52  SBS53  SBS54  SBS55  \\\n",
       "0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS56  SBS57  SBS58  SBS59  SBS60  \n",
       "0    0.0    0.0    0.0    0.0    0.0  \n",
       "1    0.0    0.0    0.0    0.0    0.0  \n",
       "2    0.0    0.0    0.0    0.0    0.0  "
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_annotations_SBS_signatures.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "b6881060-eafd-42dc-9ecb-fd11c6100e21",
   "metadata": {},
   "outputs": [],
   "source": [
    "#################################### FILTER COMBINED TABLE ####################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "63d648a2-6186-4177-a722-1751dcc94265",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows with the same icgc_donor_id but different icgc_sample_id: 182\n"
     ]
    }
   ],
   "source": [
    "# Count the number of rows with the same icgc_donor_id but different icgc_sample_id\n",
    "count_rows = combined_annotations_SBS_signatures.groupby('icgc_donor_id')['icgc_sample_id'].transform('nunique').gt(1).sum()\n",
    "print(\"Number of rows with the same icgc_donor_id but different icgc_sample_id:\", count_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "9e9545c6-81b7-4d5f-a595-0d9fcc9611bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows with the same icgc_donor_id but different icgc_sample_id: 0\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(2658, 75)"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter to keep only the sample with the highest purity per donor\n",
    "# First, sort the DataFrame by icgc_donor_id and purity (in descending order)\n",
    "combined_table_sorted = combined_annotations_SBS_signatures.sort_values(by=['icgc_donor_id', 'purity'], ascending=[True, False])\n",
    "# Then, drop duplicates, keeping the first occurrence (which will have the highest purity due to sorting)\n",
    "combined_table_filtered = combined_table_sorted.drop_duplicates(subset='icgc_donor_id', keep='first')\n",
    "\n",
    "count_rows = combined_table_filtered.groupby('icgc_donor_id')['icgc_sample_id'].transform('nunique').gt(1).sum()\n",
    "print(\"Number of rows with the same icgc_donor_id but different icgc_sample_id:\", count_rows)\n",
    "\n",
    "combined_table_filtered.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "ac009294-e15e-4f61-9f83-575442d5699e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1876, 75)"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Further filter to keep rows with purity greater than 0.5\n",
    "combined_table_filtered = combined_table_filtered[combined_table_filtered['purity'] > 0.5]\n",
    "combined_table_filtered.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d198ad3a-302a-4294-8ca7-3f8d21d0686e",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'combined_table_filtered' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mcombined_table_filtered\u001b[49m\u001b[38;5;241m.\u001b[39mto_csv(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mcombined_table_filtered.csv\u001b[39m\u001b[38;5;124m'\u001b[39m, index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'combined_table_filtered' is not defined"
     ]
    }
   ],
   "source": [
    "combined_table_filtered.to_csv('../genomic_data/combined_table_filtered.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "4138394d-8546-4d8b-ac9b-8cc06c7e55aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_table_filtered = pd.read_csv('../genomic_data/combined_table_filtered.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "1fae058e-884e-4953-aa94-4c10ca899bf5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_tumor_wgs_icgc_specimen_id</th>\n",
       "      <th>samplename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>purity</th>\n",
       "      <th>histology_abbreviation</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>Cancer Types</th>\n",
       "      <th>Sample Names</th>\n",
       "      <th>Accuracy</th>\n",
       "      <th>SBS1</th>\n",
       "      <th>SBS2</th>\n",
       "      <th>SBS3</th>\n",
       "      <th>SBS4</th>\n",
       "      <th>SBS5</th>\n",
       "      <th>SBS6</th>\n",
       "      <th>SBS7a</th>\n",
       "      <th>SBS7b</th>\n",
       "      <th>SBS7c</th>\n",
       "      <th>SBS7d</th>\n",
       "      <th>SBS8</th>\n",
       "      <th>SBS9</th>\n",
       "      <th>SBS10a</th>\n",
       "      <th>SBS10b</th>\n",
       "      <th>SBS11</th>\n",
       "      <th>SBS12</th>\n",
       "      <th>SBS13</th>\n",
       "      <th>SBS14</th>\n",
       "      <th>SBS15</th>\n",
       "      <th>SBS16</th>\n",
       "      <th>SBS17a</th>\n",
       "      <th>SBS17b</th>\n",
       "      <th>SBS18</th>\n",
       "      <th>SBS19</th>\n",
       "      <th>SBS20</th>\n",
       "      <th>SBS21</th>\n",
       "      <th>SBS22</th>\n",
       "      <th>SBS23</th>\n",
       "      <th>SBS24</th>\n",
       "      <th>SBS25</th>\n",
       "      <th>SBS26</th>\n",
       "      <th>SBS27</th>\n",
       "      <th>SBS28</th>\n",
       "      <th>SBS29</th>\n",
       "      <th>SBS30</th>\n",
       "      <th>SBS31</th>\n",
       "      <th>SBS32</th>\n",
       "      <th>SBS33</th>\n",
       "      <th>SBS34</th>\n",
       "      <th>SBS35</th>\n",
       "      <th>SBS36</th>\n",
       "      <th>SBS37</th>\n",
       "      <th>SBS38</th>\n",
       "      <th>SBS39</th>\n",
       "      <th>SBS40</th>\n",
       "      <th>SBS41</th>\n",
       "      <th>SBS42</th>\n",
       "      <th>SBS43</th>\n",
       "      <th>SBS44</th>\n",
       "      <th>SBS45</th>\n",
       "      <th>SBS46</th>\n",
       "      <th>SBS47</th>\n",
       "      <th>SBS48</th>\n",
       "      <th>SBS49</th>\n",
       "      <th>SBS50</th>\n",
       "      <th>SBS51</th>\n",
       "      <th>SBS52</th>\n",
       "      <th>SBS53</th>\n",
       "      <th>SBS54</th>\n",
       "      <th>SBS55</th>\n",
       "      <th>SBS56</th>\n",
       "      <th>SBS57</th>\n",
       "      <th>SBS58</th>\n",
       "      <th>SBS59</th>\n",
       "      <th>SBS60</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SA6125</td>\n",
       "      <td>SP2143</td>\n",
       "      <td>f393ba16-9361-5df4-e040-11ac0d4844e8</td>\n",
       "      <td>DO1000</td>\n",
       "      <td>0.562</td>\n",
       "      <td>Breast-DCIS</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>Breast-DCIS</td>\n",
       "      <td>SP2143</td>\n",
       "      <td>0.956</td>\n",
       "      <td>267.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>471.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1395.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SA6131</td>\n",
       "      <td>SP2145</td>\n",
       "      <td>f393bafd-1baa-e5f4-e040-11ac0d48450b</td>\n",
       "      <td>DO1002</td>\n",
       "      <td>0.769</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>SP2145</td>\n",
       "      <td>0.973</td>\n",
       "      <td>238.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>764.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>381.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4190.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SA6137</td>\n",
       "      <td>SP2147</td>\n",
       "      <td>f393bafe-7503-5c45-e040-11ac0d484511</td>\n",
       "      <td>DO1004</td>\n",
       "      <td>0.580</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>SP2147</td>\n",
       "      <td>0.974</td>\n",
       "      <td>373.0</td>\n",
       "      <td>474.0</td>\n",
       "      <td>6709.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2634.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>531.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_sample_id matched_tumor_wgs_icgc_specimen_id  \\\n",
       "0         SA6125                             SP2143   \n",
       "1         SA6131                             SP2145   \n",
       "2         SA6137                             SP2147   \n",
       "\n",
       "                             samplename icgc_donor_id  purity  \\\n",
       "0  f393ba16-9361-5df4-e040-11ac0d4844e8        DO1000   0.562   \n",
       "1  f393bafd-1baa-e5f4-e040-11ac0d48450b        DO1002   0.769   \n",
       "2  f393bafe-7503-5c45-e040-11ac0d484511        DO1004   0.580   \n",
       "\n",
       "  histology_abbreviation cancer_type    Cancer Types Sample Names  Accuracy  \\\n",
       "0            Breast-DCIS        BRCA     Breast-DCIS       SP2143     0.956   \n",
       "1         Breast-AdenoCA        BRCA  Breast-AdenoCA       SP2145     0.973   \n",
       "2         Breast-AdenoCA        BRCA  Breast-AdenoCA       SP2147     0.974   \n",
       "\n",
       "    SBS1   SBS2    SBS3  SBS4    SBS5  SBS6  SBS7a  SBS7b  SBS7c  SBS7d  SBS8  \\\n",
       "0  267.0    0.0     0.0   0.0   471.0   0.0    0.0    0.0    0.0    0.0   0.0   \n",
       "1  238.0  336.0     0.0   0.0   764.0   0.0    0.0    0.0    0.0    0.0   0.0   \n",
       "2  373.0  474.0  6709.0   0.0  2634.0   0.0    0.0    0.0    0.0    0.0   0.0   \n",
       "\n",
       "   SBS9  SBS10a  SBS10b  SBS11  SBS12  SBS13  SBS14  SBS15  SBS16  SBS17a  \\\n",
       "0   0.0     0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0     0.0   \n",
       "1   0.0     0.0     0.0    0.0    0.0  381.0    0.0    0.0    0.0     0.0   \n",
       "2   0.0     0.0     0.0    0.0    0.0  531.0    0.0    0.0    0.0     0.0   \n",
       "\n",
       "   SBS17b   SBS18  SBS19  SBS20  SBS21  SBS22  SBS23  SBS24  SBS25  SBS26  \\\n",
       "0     0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1     0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2     0.0  1279.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS27  SBS28  SBS29  SBS30  SBS31  SBS32  SBS33  SBS34  SBS35  SBS36  \\\n",
       "0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS37  SBS38  SBS39   SBS40  SBS41  SBS42  SBS43  SBS44  SBS45  SBS46  \\\n",
       "0    0.0    0.0    0.0  1395.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1    0.0    0.0    0.0  4190.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2    0.0    0.0    0.0     0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS47  SBS48  SBS49  SBS50  SBS51  SBS52  SBS53  SBS54  SBS55  SBS56  \\\n",
       "0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "1    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "2    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   \n",
       "\n",
       "   SBS57  SBS58  SBS59  SBS60  \n",
       "0    0.0    0.0    0.0    0.0  \n",
       "1    0.0    0.0    0.0    0.0  \n",
       "2    0.0    0.0    0.0    0.0  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_table_filtered.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "718415f9-b027-47f5-8749-074bddff317f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#################################### ACCESS, COMBINE? AND FILTER ACTUAL MUTATION FILES ####################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4ecbb792-4ae8-4598-94d4-221082355820",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gzip\n",
    "\n",
    "# Path to the VCF file\n",
    "snv_path = '/omics/groups/OE0606/internal/shared/icgc_pcawg/variant_calls/final_consensus_12oct_passonly/snv_mnv'\n",
    "snv_file_name = 'fd163b56-1c2d-4e63-9fad-ba3221ae274b.consensus.20160830.somatic.snv_mnv.vcf.gz'\n",
    "snv_file_path = f\"{snv_path}/{snv_file_name}\"\n",
    "\n",
    "# Find the index of the last header line (starting with a single '#')\n",
    "header_line = -1\n",
    "with gzip.open(snv_file_path, 'rt') as file:  # Open as text with gzip handling\n",
    "    for i, line in enumerate(file):\n",
    "        if line.startswith('#'):\n",
    "            header_line = i\n",
    "\n",
    "# Use the identified header line to read the VCF data into a DataFrame\n",
    "snv_df = pd.read_csv(snv_file_path, compression='gzip', header=header_line, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b660ccb7-463a-4d2a-b71d-412b600735db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>5843414</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7750293</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11914490</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Callers=broad,muse,sanger;NumCallers=3;VAF=0.1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CHROM       POS ID REF ALT QUAL FILTER  \\\n",
       "0      1   5843414  .   G   T    .      .   \n",
       "1      1   7750293  .   G   A    .      .   \n",
       "2      1  11914490  .   A   T    .      .   \n",
       "\n",
       "                                                INFO  \n",
       "0  Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...  \n",
       "1  Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...  \n",
       "2  Callers=broad,muse,sanger;NumCallers=3;VAF=0.1...  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv_df.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "bfbfd7b3-8531-4590-b83e-40fe948b4405",
   "metadata": {},
   "outputs": [],
   "source": [
    "snv_df=snv_df[['#CHROM', 'POS', 'REF', 'ALT', 'INFO']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "20c97816-9e30-4a1a-9945-7ad59e2f5bd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>5843414</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7750293</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11914490</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Callers=broad,muse,sanger;NumCallers=3;VAF=0.1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CHROM       POS REF ALT                                               INFO\n",
       "0      1   5843414   G   T  Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...\n",
       "1      1   7750293   G   A  Callers=broad,dkfz,muse,sanger;NumCallers=4;VA...\n",
       "2      1  11914490   A   T  Callers=broad,muse,sanger;NumCallers=3;VAF=0.1..."
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv_df.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "58c2094e-879d-496f-b298-302cb16a202c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the regular expression for extracting the needed information\n",
    "pattern = r'VAF=(?P<VAF>\\d+\\.\\d+);.*t_alt_count=(?P<t_alt_count>\\d+);t_ref_count=(?P<t_ref_count>\\d+);Variant_Classification=(?P<Variant_Classification>[^;]+)'\n",
    "\n",
    "# Extract the data using the regex pattern\n",
    "info_extracted = snv_df['INFO'].str.extract(pattern)\n",
    "\n",
    "# Convert extracted data to numeric where necessary (for VAF, t_alt_count, and t_ref_count)\n",
    "info_extracted['VAF'] = pd.to_numeric(info_extracted['VAF'])\n",
    "info_extracted['t_alt_count'] = pd.to_numeric(info_extracted['t_alt_count'])\n",
    "info_extracted['t_ref_count'] = pd.to_numeric(info_extracted['t_ref_count'])\n",
    "\n",
    "# Add the extracted columns to the original DataFrame\n",
    "snv_small = pd.concat([snv_df, info_extracted], axis=1)\n",
    "\n",
    "# Optionally, drop the original 'INFO' column if no longer needed\n",
    "snv_small.drop('INFO', axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "f5e24afa-7201-404d-b6ae-48fe702789c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAF</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>Variant_Classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>5843414</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.2973</td>\n",
       "      <td>11.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>IGR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7750293</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>8.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11914490</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.1429</td>\n",
       "      <td>5.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>IGR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CHROM       POS REF ALT     VAF  t_alt_count  t_ref_count  \\\n",
       "0      1   5843414   G   T  0.2973         11.0         26.0   \n",
       "1      1   7750293   G   A  0.2500          8.0         24.0   \n",
       "2      1  11914490   A   T  0.1429          5.0         30.0   \n",
       "\n",
       "  Variant_Classification  \n",
       "0                    IGR  \n",
       "1                 Intron  \n",
       "2                    IGR  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv_small.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "0aa26d2c-007f-4e2b-be6a-f7fbb8a80b47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1141, 8)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv_small.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "82696c2a-92bc-47aa-8743-885bb562df67",
   "metadata": {},
   "outputs": [],
   "source": [
    "def snv_gz_to_csv(snv_file_name, verbose=False):\n",
    "    \n",
    "    snv_file_path = f\"{snv_path}/{snv_file_name}\"\n",
    "\n",
    "    # Find the index of the last header line (starting with a single '#')\n",
    "    header_line = -1\n",
    "    with gzip.open(snv_file_path, 'rt') as file:  # Open as text with gzip handling\n",
    "        for i, line in enumerate(file):\n",
    "            if line.startswith('#'):\n",
    "                header_line = i\n",
    "    \n",
    "    # Use the identified header line to read the VCF data into a DataFrame\n",
    "    snv_df = pd.read_csv(snv_file_path, compression='gzip', header=header_line, sep='\\t', dtype={0: str})\n",
    "    if verbose:\n",
    "        print('Original data', '\\n',  snv_df.head(3), '\\n\\n\\n\\n')\n",
    "    snv_df = snv_df[['#CHROM', 'POS', 'REF', 'ALT', 'INFO']]\n",
    "    if verbose:\n",
    "        print('Relevant columns', '\\n',snv_df.head(3),'\\n\\n\\n\\n')\n",
    "\n",
    "\n",
    "    # Check if 'REF' or 'ALT' have more than one character\n",
    "    condition = (snv_df['REF'].str.len() > 1) | (snv_df['ALT'].str.len() > 1)\n",
    "    # Count the number of such rows\n",
    "    count_multichars = condition.sum()\n",
    "    if count_multichars!=0:\n",
    "        print(f\"Number of rows with multiple characters in 'REF' or 'ALT': {count_multichars}\")\n",
    "    # Drop the rows where 'REF' or 'ALT' have more than one character\n",
    "    snv_df = snv_df[~condition]\n",
    "\n",
    "    # Define the regular expression for extracting the needed information\n",
    "    pattern = r'VAF=(?P<VAF>\\d+\\.\\d+);.*t_alt_count=(?P<t_alt_count>\\d+);t_ref_count=(?P<t_ref_count>\\d+);Variant_Classification=(?P<Variant_Classification>[^;]+)'\n",
    "\n",
    "    # Extract the data using the regex pattern\n",
    "    info_extracted = snv_df['INFO'].str.extract(pattern)\n",
    "\n",
    "    # Convert extracted data to numeric where necessary (for VAF, t_alt_count, and t_ref_count)\n",
    "    info_extracted['VAF'] = pd.to_numeric(info_extracted['VAF'])\n",
    "    info_extracted['t_alt_count'] = pd.to_numeric(info_extracted['t_alt_count'])\n",
    "    info_extracted['t_ref_count'] = pd.to_numeric(info_extracted['t_ref_count'])\n",
    "\n",
    "    # Add the extracted columns to the original DataFrame\n",
    "    snv_small = pd.concat([snv_df, info_extracted], axis=1)\n",
    "    \n",
    "    # Optionally, drop the original 'INFO' column if no longer needed\n",
    "    snv_small.drop('INFO', axis=1, inplace=True)\n",
    "    if verbose:\n",
    "        print('Extracted info', '\\n',snv_small.head(3))\n",
    "    return snv_small"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "7a4476a3-761c-4a3e-9a00-d47776038b7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "gz_path  = '/omics/groups/OE0606/internal/shared/icgc_pcawg/variant_calls/final_consensus_12oct_passonly/snv_mnv'\n",
    "csv_path = '/omics/groups/OE0606/internal/jonas/genomic_data/snv_variant_calls'\n",
    "\n",
    "for filename in os.listdir(gz_path):\n",
    "    if filename.endswith(\".vcf.gz\"):\n",
    "        csv_filename = filename.replace('.consensus.20160830.somatic.snv_mnv.vcf.gz', '.csv')\n",
    "        csv_filepath = os.path.join(csv_path, csv_filename)\n",
    "        \n",
    "        # Convert VCF to CSV using your existing function\n",
    "        snv_df = snv_gz_to_csv(filename)\n",
    "        \n",
    "        # Save the DataFrame to a CSV file\n",
    "        snv_df.to_csv(csv_filepath, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5dd64be8-8ebe-4b59-8267-38a0aaa6bc28",
   "metadata": {},
   "outputs": [],
   "source": [
    "## This generated 1902 files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "7b55058c-5002-4e95-a630-f83749304350",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of files in the directory not in the DataFrame: 700\n",
      "Number of samples in the DataFrame not in the directory files: 674\n",
      "Number of shared files: 1202\n"
     ]
    }
   ],
   "source": [
    "## sort out files for samples that didnt make it through preprocessing (are not in combined_table_filtered)\n",
    "\n",
    "sample_names = combined_table_filtered['samplename']\n",
    "\n",
    "# List filenames in the directory\n",
    "file_names = [os.path.splitext(file)[0] for file in os.listdir(csv_path) if os.path.isfile(os.path.join(csv_path, file))]\n",
    "\n",
    "# Convert to sets for efficient comparison\n",
    "file_names_set = set(file_names)\n",
    "sample_names_set = set(sample_names)\n",
    "\n",
    "# Samples not in files\n",
    "samples_not_in_files = sample_names_set.difference(file_names_set)\n",
    "\n",
    "# Files not in DataFrame\n",
    "files_not_in_df = file_names_set.difference(sample_names_set)\n",
    "\n",
    "shared_files = file_names_set.intersection(sample_names_set)\n",
    "\n",
    "# Display the results\n",
    "print(f\"Number of files in the directory not in the DataFrame: {len(files_not_in_df)}\")\n",
    "print(f\"Number of samples in the DataFrame not in the directory files: {len(samples_not_in_files)}\")\n",
    "print(f\"Number of shared files: {len(shared_files)}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "f2ee2b78-a9b1-47ab-8fc7-0b4977280030",
   "metadata": {},
   "outputs": [],
   "source": [
    "# store list of files which have data in csv file stored (already filtered)\n",
    "with open('/omics/groups/OE0606/internal/jonas/genomic_data/shared_files.txt', 'w') as file:\n",
    "    for item in shared_files:\n",
    "        file.write(f\"{item}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64d7d214-55fb-4681-9e0f-cc5df7ac04e6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
